File: Physics/Code/Introduction_Fortran/set3.py
import matplotlib.pyplot as plt
import numpy as np

#unload data
data = np.loadtxt('/Users/phihung/Documents/PHY/first/homework/hw_01/gregory_convergence.dat')

# unpack value
n = data[:,0]
pi = data[:,1]

#define y=pi function
def pi_function(n):
    return n*0 + np.pi
# record on k
k = pi_function(n)

# plot the repetition vs 
plt.plot(n,pi,label='Gregory series')
plt.xlabel('N [repetitions]')
plt.ylabel('Value [arb. unit]')

plt.plot(n,k, color='red', label='pi')

plt.title(r'Gregory Series')

plt.legend()
plt.grid(True)

plt.show()

File: Physics/Code/Introduction_Fortran/set3.f90
implicit none

integer, parameter :: n_max = 500  ! N values
integer :: n_2, n_1
double precision :: value ! the pi values recorded

! open the file to record
open(file='/Users/phihung/Documents/PHY/first/homework/hw_01/gregory_convergence.dat', unit=11)
do n_1=1, n_max !calculate convergence from N = 1 to 500
    value = 0.0 ! reset value each loop
    do n_2=1, n_1 !calculate the series for each N value
        value = value + (((-1)**(float(n_2)+1)) / (2*float(n_2) - 1)) !gregory series
    end do 
    value = value * 4.0 ! times 4 from outside
    write(11,*) n_1, value ! write the N and the pi value recorded
end do
close(unit=11)

end